tutorials/create-cloud-build-image-factory-using-packer/index.md

```diff
@@ -41,59 +59,297 @@ gcloud config set compute/zone $ZONE
+<details>
+<summary>Click to reveal commands</summary>
+
+```sh
+gcloud services enable sourcerepo.googleapis.com
+gcloud services enable cloudapis.googleapis.com
+gcloud services enable compute.googleapis.com
+gcloud services enable servicemanagement.googleapis.com
+gcloud services enable storage-api.googleapis.com
+gcloud services enable cloudbuild.googleapis.com
+```
+</details>
+
+## Building a Windows Image
+
+<details>
+<summary>Click to reveal commands</summary>
+
+```sh
+gcloud services enable sourcerepo.googleapis.com
+gcloud services enable cloudapis.googleapis.com
+gcloud services enable compute.googleapis.com
+gcloud services enable servicemanagement.googleapis.com
+gcloud services enable storage-api.googleapis.com
+gcloud services enable cloudbuild.googleapis.com
+gcloud services enable secretmanager.googleapis.com
+```
+</details>
+
+# Task 3a: Managing secrets for params using the Google Secrets Manager (Windows Image Only)
+In this section, we will be using the [Google Secrets Manager](https://cloud.google.com/secret-manager) to store our input values for packer in a secure and modular way. Although it's easier to simply hard code parameters into the packer template file (eg. network tags, source GCE image), using a central source of truth like a secrets manager increases manageability and reuseability amongst teams.
+
+Create your secrets using the following commands. Optionally, you can customize the values using the [documentation](https://cloud.google.com/secret-manager/docs/creating-and-accessing-secrets)
+
+<details>
+<summary>Click to reveal commands</summary>
+
+```sh
+echo -n "windows-2019" | gcloud secrets create image_factory-image_family --replication-policy="automatic" --data-file=-
+
+echo -n "golden-windows" | gcloud secrets create image_factory-image_name --replication-policy="automatic" --data-file=-
+
+echo -n "n1-standard-1" | gcloud secrets create image_factory-machine_type --replication-policy="automatic" --data-file=-
+
+echo -n "australia-southeast1" | gcloud secrets create image_factory-region --replication-policy="automatic" --data-file=-
+
+echo -n "australia-southeast1-b" | gcloud secrets create image_factory-zone --replication-policy="automatic" --data-file=-
+
+echo -n "default" | gcloud secrets create image_factory-network --replication-policy="automatic" --data-file=-
+
+echo -n "packer-win" | gcloud secrets create image_factory-tags --replication-policy="automatic" --data-file=-
+```
+
+</details>
+
+# Task 3b: Create a new VPC firewall to allow winrm for packer (Windows Image Only)
+Before you can provision using the winrm communicator, you need to allow traffic through google's firewall on the winrm port (tcp:5986). This creates a new firewall called `packer-win` that is stored in Secrets Manager (Task 3a) and used by Cloud Build in the cloudbuild.yaml config file. 
+
+<details>
+<summary>Click to reveal commands</summary>
+
+```sh
+gcloud compute firewall-rules create packer-win --allow tcp:5986
```

Comment:
Restricting by network tag would be good.

Also, consider a naming scheme like `allow-winrm-ingress-to-packer`

Reply:
It appears that half of this suggestion was implemented.

Reply:
updated